pkgs/applications/science/astronomy/siril/default.nix

```diff
@@ -0,0 +1,74 @@
+{ stdenv,
+  fetchgit,
+  pkg-config,
+  meson,
+  ninja,
+  cmake,
+  git,
+  gtk3,
+  wrapGAppsHook,
+  fftwFloat,
+  cfitsio,
+  gsl,
+  exiv2,
+  opencv,
+  libconfig,
+  libraw,
+  libtiff,
+  libpng,
+  libjpeg,
+  ffmpeg,
+  gnuplot,
+  curl
+}:
+
+
+stdenv.mkDerivation rec {
+  pname = "siril";
+  version = "0.99.6";
+
+  src = fetchgit {
```

Comment:
Can you please use fetchFromGitLab?

Reply:
Is there a way to retrieve submodules with fetchFromGitLab? I originally had it that way but it fails to pull some out of tree libraries (that are specific to this piece of software so I feel it would be a mistake to package them independently).

Reply:
Fixed by adding submodule dep as another derivation. I realized the submodule is 3rd party, and used by other projects so mine as well make it into a derivation.